plugins/modules/source_control/github/github_tag.py

```diff
@@ -0,0 +1,272 @@
+                         details="Please check username and password or token "
+                                 "for repository %s" % module.params.get('repo'))
+
+    return client
+
+
+def get_repo(module, client):
+    '''
+    Check if tag already exists and return repository object
+    '''
+
+    org = module.params.get('organization')
+    repo = module.params.get('repo')
+    tag = module.params.get('tag')
+
+    try:
+        repository = client.repository(org, repo)
+    except (github3.exceptions.ClientError, github3.exceptions.NotFoundError):
+        module.fail_json(msg="Repository %s/%s doesn't exist" % (org, repo))
+
+    tag_exists = False
+
+    for tag_item in repository.tags():
+        if tag_item.name == tag:
+            tag_exists = True
+
+    if tag_exists:
+        return False
+
+    return repository
+
+
+def create_tag(module, repository):
+    '''
+    Create annotated git tag
+    '''
+
+    branch = module.params.get('branch')
+    tag = module.params.get('tag')
+    tagger = module.params.get('tagger')
+    description = module.params.get('description')
+
+    latest_commit = repository.commit(sha=branch)
+
+    # Currently the github3.py library requires the date to be set,
+    # eventhough not required by API.
+    current_date = datetime.now()
+    tagger['date'] = current_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+    tag_params = dict(
+        tag=tag,
+        tagger=tagger,
+        sha=latest_commit.sha,
+        message=description,
+        obj_type="commit"
+    )
+
+    try:
+        new_tag = repository.create_tag(**tag_params)
+    except (github3.exceptions.ClientError, github3.exceptions.UnprocessableEntity) as err:
+        module.fail_json(
+            msg='Failed to create a annotated tag with the following error: %s' % to_native(err))
+
+    return new_tag
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            url=dict(type='str'),
+            username=dict(type='str'),
+            password=dict(no_log=True),
+            token=dict(no_log=True),
+            organization=dict(required=True, type='str'),
+            repo=dict(required=True),
+            branch=dict(required=True, type='str'),
+            tag=dict(required=True, type='str'),
+            tagger=dict(required=True, type='dict'),
+            description=dict(required=True, type='str')
+        ),
+        required_together=[('username', 'password')],
+        required_one_of=[('password', 'token')],
+        mutually_exclusive=[('password', 'token')]
+    )
+
+    if not HAS_GITHUB3_PACKAGE:
+        module.fail_json(msg=missing_required_lib('github3.py >= 1.0.0'),
+                         exception=GITHUB_IMP_ERR)
+
+    client = authenticate(module)
+    repository = get_repo(module, client)
+
+    if not repository:
+        module.exit_json(changed=False, msg="Tag %s already exists." % module.params.get('tag'))
```

Comment:
```suggestion
        module.exit_json(changed=False, msg="Tag %s already exists." % module.params['tag'])
```
The values are always present. (Same at the other places.)